src/wolfio.c

```diff
@@ -989,6 +989,21 @@ int wolfIO_Recv(SOCKET_T sd, char *buf, int sz, int rdFlags)
     recvd = (int)RECV_FUNCTION(sd, buf, (size_t)sz, rdFlags);
     recvd = TranslateReturnCode(recvd, (int)sd);
 
+    if (recvd < 0) {
+        int last_err = wolfSSL_LastError(recvd);
+        if ((last_err == SOCKET_EWOULDBLOCK)
+#if SOCKET_EWOULDBLOCK != SOCKET_EAGAIN
+            || (last_err == SOCKET_EAGAIN)
+#endif
+#ifdef SOCKET_ETIMEDOUT
+            || (last_err == SOCKET_ETIMEDOUT)
+#endif
+            )
+        {
+            return SOCKET_NODATA;
+        }
+    }
+
     return recvd;
 }
```

Comment:
This is handled in `EmbedReceive` in `TranslateIoError`. I don't think it is necessary in this PR.

Reply:
fixed by refactor